src/dtflw/databricks.py

```diff
@@ -1,61 +1,203 @@
+from datetime import timedelta
 from pyspark.sql import SparkSession
 
+class _Singelton:
+    def __init__(self, klass):
+        self.klass = klass
+        self.instance = None
 
-def run_notebook(path: str, timeout: int, arguments: dict):
+    def __call__(self, **kwds):
+        if self.instance == None:
+            try:
+                self.instance = self.klass(**kwds)
+            except TypeError as err:
+                raise RuntimeError(f"Please provide input parameters. {err.args}")
+
+        return self.instance
+
+
+def get_spark_session(
+    host: str = None,
+    cluster_id: str = None,
+    token: str = None,
+) -> SparkSession:
     """
-    Runs the current notebook.
+    Parameters
+    ----------
+    host : str, optional
+        The Databricks workspace URL
+    cluster_id: str, optional
+        The cluster identifier where the Databricks connect queries should be executed.
+    token: str, optional
+        The Databricks personal access token used to authenticate into the cluster and on
+        whose behalf the queries are executed.
+
+    Returns
+    -------
+    SparkSession instance.
     """
-    return get_dbutils().notebook.run(path, timeout, arguments)
+    session = SparkSession.getActiveSession()
```

Comment:
we can just return the session if it's available here.
Also, let's raise an exception in case cluster_id, host, or token are not provided.

Reply:
This is the signature of the method
`def getActiveSession(cls) -> Optional["SparkSession"]:`
so, it could return **None**, and that is why I store the result and do the session creation in case if there is no session.